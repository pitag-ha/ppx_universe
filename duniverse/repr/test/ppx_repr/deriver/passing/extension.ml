type 'a typ = 'a Repr.t

module Simple = struct
  let (_ : (int * string) list typ) = [%typ: (int * string) list]
end

module Alias = struct
  type t = unit

  let t = Repr.unit
  let (_ : unit typ) = [%typ: t]
end

module Sum = struct
  let (_ : [ `Foo | `Bar of string ] typ) = [%typ: [ `Foo | `Bar of string ]]
end

module Params = struct
  let __ : type a. a typ -> a list typ = [%typ: 'a list]
  let __ : type a b. a typ -> b typ -> (a * b * a) typ = [%typ: 'a * _ * 'a]
  let __ : type a b. a typ -> b typ -> (a, b) result typ = [%typ: (_, _) result]
end

module Namespace = struct
  let (_ : string typ) = [%repr.typ: string]
end
