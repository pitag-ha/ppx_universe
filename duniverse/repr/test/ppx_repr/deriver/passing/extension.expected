type 'a typ = 'a Repr.t
module Simple =
  struct
    let (_ : (int * string) list typ) =
      Repr.list (Repr.pair Repr.int Repr.string)
  end
module Alias =
  struct type t = unit
         let t = Repr.unit
         let (_ : unit typ) = t end
module Sum =
  struct
    let (_ : [ `Foo  | `Bar of string ] typ) =
      Repr.sealv
        (Repr.(|~)
           (Repr.(|~)
              (Repr.variant "t"
                 (fun foo ->
                    fun bar -> function | `Foo -> foo | `Bar x1 -> bar x1))
              (Repr.case0 "Foo" `Foo))
           (Repr.case1 "Bar" Repr.string (fun x1 -> `Bar x1)))
  end
module Params =
  struct
    let __ : type a. a typ -> a list typ = fun a -> Repr.list a
    let __ : type a b. a typ -> b typ -> (a * b * a) typ =
      fun a -> fun _x__001_ -> Repr.triple a _x__001_ a
    let __ : type a b. a typ -> b typ -> (a, b) result typ =
      fun _x__002_ -> fun _x__003_ -> Repr.result _x__002_ _x__003_
  end
module Namespace = struct let (_ : string typ) = Repr.string end
